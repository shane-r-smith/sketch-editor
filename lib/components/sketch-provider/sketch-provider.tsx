import { Provider } from "jotai";
import { type PropsWithChildren } from "react";
import { sketchStore } from "../../sketch.store";

export function SketchProvider({ children }: PropsWithChildren) {
  return <Provider store={sketchStore}>{children}</Provider>;
}
