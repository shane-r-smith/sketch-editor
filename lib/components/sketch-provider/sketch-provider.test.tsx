import { render } from "@testing-library/react";
import { SketchProvider } from "./sketch-provider";

describe("<SketchProvider />", () => {
  it("renders children correctly", () => {
    const { getByText } = render(
      <SketchProvider>
        <div>Test Child</div>
      </SketchProvider>
    );
    expect(getByText("Test Child")).toMatchSnapshot();
  });
});
