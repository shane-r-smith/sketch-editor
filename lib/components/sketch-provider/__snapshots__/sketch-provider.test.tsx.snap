// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SketchProvider /> > renders children correctly 1`] = `
<div>
  Test Child
</div>
`;
